src/test/scala/mesosphere/marathon/integration/setup/MarathonCallbackTestSupport.scala

```diff
@@ -52,4 +52,33 @@ trait MarathonCallbackTestSupport extends ExternalMarathonIntegrationTest {
     }
   }
 
+  /**
+    * Wait for the events of the given kinds (=types).
+    */
+  def waitForEvents(kinds: String*)(maxWait: FiniteDuration = 30.seconds): Map[String, Seq[CallbackEvent]] = {
+
+    val deadline = maxWait.fromNow
+
+    /** Receive the events for the given kinds (duplicates allowed) in any order. */
+    val receivedEventsForKinds: Seq[CallbackEvent] = {
+      var eventsToWaitFor = kinds
+      var receivedEvents: Vector[CallbackEvent] = Vector()
```

Comment:
Hmmm. Is that important? Is that code, as it is, hard to understand for you? Or what is the problem you see with it?